sgkit/variables.py

```diff
@@ -0,0 +1,225 @@
+from dataclasses import dataclass
+from typing import Dict, Hashable, Mapping, Set, Union, overload
+
+import xarray as xr
+
+
+@dataclass(frozen=True)
+class Spec:
+    """Root type Spec"""
+
+    default_name: str
+
+
+@dataclass(frozen=True)
+class ArrayLikeSpec(Spec):
+    """ArrayLike type spec"""
+
+    kind: Union[None, str, Set[str]] = None
+    ndim: Union[None, int, Set[int]] = None
+
+
+call_genotype = ArrayLikeSpec("call_genotype", kind="i", ndim=3)
```

Comment:
We should decide if we want to list this in a single namespace, group these, and or what order do we want them documented. Also would be good to have bidirectional links between variables and methods.

Reply:
+1 to making it possible to see what methods use or produce a variable.

Reply:
I agree with @tomwhite here